packages/react-next/src/components/Shimmer/Shimmer.base.tsx

```diff
@@ -1,126 +1,94 @@
 import * as React from 'react';
-import {
-  classNamesFunction,
-  DelayedRender,
-  getNativeProps,
-  divProperties,
-  Async,
-  initializeComponentRef,
-} from '../../Utilities';
+import { classNamesFunction, DelayedRender, getNativeProps, divProperties } from '../../Utilities';
 import { IShimmerProps, IShimmerStyleProps, IShimmerStyles } from './Shimmer.types';
 import { ShimmerElementsGroup } from './ShimmerElementsGroup/ShimmerElementsGroup';
-
-export interface IShimmerState {
-  /**
-   * Flag for knowing when to remove the shimmerWrapper from the DOM.
-   */
-  contentLoaded?: boolean;
-}
+import { useSetTimeout, useConst } from '@uifabric/react-hooks';
 
 const TRANSITION_ANIMATION_INTERVAL = 200; /* ms */
+const COMPONENT_NAME = 'Shimmer';
 
 const getClassNames = classNamesFunction<IShimmerStyleProps, IShimmerStyles>();
 
 /**
  * {@docCategory Shimmer}
  */
-export class ShimmerBase extends React.Component<IShimmerProps, IShimmerState> {
-  public static defaultProps: IShimmerProps = {
-    isDataLoaded: false,
-  };
-
-  private _classNames: { [key in keyof IShimmerStyles]: string };
-  private _lastTimeoutId: number;
-  private _async: Async;
+export const ShimmerBase = React.forwardRef<HTMLDivElement, IShimmerProps>((props, ref) => {
+  const {
+    styles,
+    shimmerElements,
+    children,
+    width,
+    className,
+    customElementsGroup,
+    theme,
+    ariaLabel,
+    shimmerColors,
+    isDataLoaded = false,
+  } = props;
+
+  const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties);
+
+  const classNames: { [key in keyof IShimmerStyles]: string } = getClassNames(styles!, {
+    theme: theme!,
+    isDataLoaded,
+    className,
+    transitionAnimationInterval: TRANSITION_ANIMATION_INTERVAL,
+    shimmerColor: shimmerColors && shimmerColors.shimmer,
+    shimmerWaveColor: shimmerColors && shimmerColors.shimmerWave,
+  });
+
+  const internalState = useConst<{ lastTimeoutId: number }>(() => ({
+    lastTimeoutId: 0,
+  }));
```

Comment:
You can remove the wrapper here.
```suggestion
  const internalState = useConst({
    lastTimeoutId: 0,
  });
```